message.go

```diff
@@ -83,6 +84,40 @@ func (m message) MarshalJSON() ([]byte, error) {
 	return m.json, nil
 }
 
+// setSentAt remarshalls the message with a new sentAt timestamp
+func (m *message) setSentAt(ts time.Time, maxBytes int) (err error) {
+	switch msg := m.msg.(type) {
+	case Alias:
+		msg.SentAt = ts
+		m.msg = msg
+	case Group:
+		msg.SentAt = ts
+		m.msg = msg
+	case Identify:
+		msg.SentAt = ts
+		m.msg = msg
+	case Page:
+		msg.SentAt = ts
+		m.msg = msg
+	case Screen:
+		msg.SentAt = ts
+		m.msg = msg
+	case Track:
+		msg.SentAt = ts
+		m.msg = msg
+	default:
+		err = fmt.Errorf("messages with custom types cannot be enqueued: %T", msg)
+		return
+	}
+
+	if m.json, err = json.Marshal(m.msg); err == nil {
+		if len(m.json) > maxBytes {
+			err = ErrMessageTooBig
+		}
+	}
```

Comment:
We can avoid this like I mentioned in the other mentioned.

Reply:
I made the requested changes [666ce95](https://github.com/rudderlabs/analytics-go/pull/27/commits/666ce9581f7e1cf64de68864dd50eeceb30407d0)